portfolio/src/main/webapp/calendar.js

```diff
@@ -36,21 +41,45 @@ function createNewCalendarEvent() {
   const endTime =
       getTimeObject(document.getElementById('new-event-end-time').value);
 
-  const eventName = document.getElementById('new-event-name').value;
+  // If user enters an empty event name, sets the name to a default string.
+  const eventName =
+      document.getElementById('new-event-name').value || EVENT_DEFAULT_NAME;
 
   // Checks that end time is later than start time.
   // If the time order is wrong, show a warning message on the UI.
   // Otherwise, proceed to create a new event list element and
   // send the new event to the Calendar servlet.
   if (endTime.getTime() <= startTime.getTime()) {
-    document.getElementById('event-end-time-warning').style.visibility =
-        'visible';
+    $('#event-warning').removeClass('d-none');
+    $('#event-warning').show();
+    $('#event-warning').text('End time must be later than start time.');
   } else {
-    document.getElementById('event-end-time-warning').style.visibility =
-        'hidden';
+    $('#event-warning').hide();
+    $('#event-warning')
+
     const newCalendarEvent = new CalendarEvent(eventName, startTime, endTime);
-    updateCalendarEventList(newCalendarEvent);
-    document.getElementById('new-event-name').value = 'New Event';
+
+    // Get all the events currently displayed on the UI.
+    // Look through this set of events to ensure that duplicate events
+    // do not get added again.
+    const allEventJson = collectAllEvents();
```

Comment:
nit: Does this actually contain JSON? I think they're just instances of `CalendarEvent` aren't they?

Reply:
Yes, this array now only cotnains instances of CalendarEvent's.